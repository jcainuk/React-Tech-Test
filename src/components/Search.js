import React, { useState } from 'react';
import '../styles/Search.css';
import PropTypes from 'prop-types';
import getImages from '../requests/getImages';

const Search = () => {
  const [value, setValue] = useState();

  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(value));
  };

  return (
    <>
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          className="search-input"
          type="text"
          onChange={(e) => setValue(e.target.value)}
          data-testid="search-input"
        />
        <button className="search-btn" type="submit" data-testid="search-btn">
          Go!
        </button>
      </form>
    </>
  );
};

Search.propTypes = {
  setSearchResults: PropTypes.func.isRequired,
};

export default Search;
