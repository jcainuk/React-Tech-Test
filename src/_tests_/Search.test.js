import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Search from '../components/Search';

describe('It renders correctly', () => {
  const validProps = {
    searchText: 'Liverpool',
    setSearchText: jest.fn(),
    onSubmit: jest.fn(),
  };
  it('renders as expected', () => {
    const { asFragment } = render(<Search {...validProps} />);
    expect(asFragment).toMatchSnapshot();
  });
  it('button fires the search event', () => {
    const { getByRole } = render(<Search {...validProps} />);
    const button = getByRole('button');
    userEvent.click(button);
    expect(validProps.onSubmit).toHaveBeenCalled();
  });
});
