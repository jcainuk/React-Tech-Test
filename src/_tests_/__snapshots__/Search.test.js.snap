// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <form
      class="search-form"
    >
      <input
        class="search-input"
        data-testid="search-input"
        placeholder="Search for images"
        type="text"
      />
      <button
        class="search-btn"
        data-testid="search-btn"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
